packages/devextreme/testing/testcafe/tests/scheduler/layout/templates/appointmentTemplate.ts

```diff
@@ -66,3 +76,274 @@ fixture.disablePageReloads`Layout:Templates:appointmentTemplate`
+  },
+  {
+    viewType: 'timelineMonth',
+    groupOrientation: 'vertical',
+    rtlEnabled: true,
+  },
+]);
+
+testOptions.forEach(({
+  viewType,
+  groupOrientation,
+  scrollMode,
+  rtlEnabled,
+}) => {
+  const currentDate = new Date(2024, 0, 2);
+  const HOUR = 1000 * 60 * 60;
+
+  const startDayHour = 8;
+  const endDayHour = 20;
+
+  let resourceCount = 30;
+
+  let appointmentDates: Date[];
+
+  if (scrollMode === 'standard') {
+    resourceCount = 10;
+  }
+
+  switch (viewType) {
+    case 'agenda': {
+      appointmentDates = Array.from(
+        { length: 12 },
+        (_, index) => {
+          const result = new Date(2024, 0, 1, 8);
+          result.setHours(result.getHours() + index);
+          return result;
+        },
+      );
+
+      break;
+    }
+    case 'week':
+    case 'timelineWeek': {
+      if (scrollMode === 'standard' && groupOrientation === 'horizontal') {
+        resourceCount = 2;
+      }
+
+      appointmentDates = Array.from(
+        { length: 12 * 7 },
+        (_, index) => {
+          const result = new Date(2023, 11, 31, 8);
+          result.setDate(result.getDate() + Math.floor(index / 12));
+          result.setHours(result.getHours() + (index % 12));
+          return result;
+        },
+      );
+
+      break;
+    }
+    case 'workWeek':
+    case 'timelineWorkWeek': {
+      if (scrollMode === 'standard' && groupOrientation === 'horizontal') {
+        resourceCount = 2;
+      }
+
+      appointmentDates = Array.from(
+        { length: 12 * 5 },
+        (_, index) => {
+          const result = new Date(2024, 0, 1, 8);
+          result.setDate(result.getDate() + Math.floor(index / 12));
+          result.setHours(result.getHours() + (index % 12));
+          return result;
+        },
+      );
+
+      break;
+    }
+    case 'month':
+    case 'timelineMonth': {
+      appointmentDates = Array.from(
+        { length: 31 },
+        (_, index) => {
+          const result = new Date(2024, 0, 1, 8);
+          result.setDate(result.getDate() + index);
+          return result;
+        },
+      );
+
+      break;
+    }
+    case 'day':
+    case 'timelineDay':
+    default: {
+      appointmentDates = Array.from(
+        { length: 12 },
+        (_, index) => {
+          const result = new Date(2024, 0, 2, 8);
+          result.setHours(result.getHours() + index);
+          return result;
+        },
+      );
+
+      break;
+    }
+  }
+
+  const datesToCheck = [
+    appointmentDates[0],
+    appointmentDates[Math.floor(appointmentDates.length / 3)],
+    appointmentDates[appointmentDates.length - 1],
+  ];
+
+  const groupsToCheck = [
+    { groupId: 0 },
+    { groupId: Math.floor(resourceCount / 3) },
+    { groupId: resourceCount - 1 },
+  ];
+
+  test(`targetedAppointmentData should be correct with groups (viewType="${viewType}", groupOrientation="${groupOrientation}", scrollMode="${scrollMode}", rtlEnabled="${rtlEnabled}") (T1205120)`, async (t) => {
```

Comment:
Optional.

Let's add a ```NOTE``` comment to this test, that it's a success if no errors in the template occurred.
It may take time to understand what going on here because this test doesn't have any ```assert/expect``` calls